tests/integration/security/filebased_tls_origination/destination_rule_tls_test.go

```diff
@@ -53,97 +55,117 @@ func TestDestinationRuleTls(t *testing.T) {
+							// so that we don't need to manage it ourselves.
+							// The paths here match the destination rule above
+							Annotations: echo.NewAnnotations().
+								Set(echo.SidecarVolume, fmt.Sprintf(`{"custom-certs":{"configMap":{"name":"%s-certs"}}}`, serverSvcName)).
+								Set(echo.SidecarVolumeMount, `{"custom-certs":{"mountPath":"/etc/certs/custom"}}`),
+						}},
+					}).
+					With(&svcs[c.Index()][1], echo.Config{
+						Service:   serverSvcName,
+						Namespace: ns,
+						Ports: []echo.Port{
+							{
+								Name:         "grpc",
+								Protocol:     protocol.GRPC,
+								InstancePort: 8090,
+								TLS:          true,
+							},
+							{
+								Name:         "http",
+								Protocol:     protocol.HTTP,
+								InstancePort: 8091,
+								TLS:          true,
+							},
+							{
+								Name:         "tcp",
+								Protocol:     protocol.TCP,
+								InstancePort: 8092,
+								TLS:          true,
+							},
 						},
-						{
-							Name:         "tcp",
-							Protocol:     protocol.TCP,
-							InstancePort: 8092,
-							TLS:          true,
-						},
-					},
-					Pilot: p,
-					// Set up TLS certs on the server. This will make the server listen with these credentials.
-					TLSSettings: &common.TLSSettings{
-						RootCert:   mustReadFile(t, "root-cert.pem"),
-						ClientCert: mustReadFile(t, "cert-chain.pem"),
-						Key:        mustReadFile(t, "key.pem"),
-						// Override hostname to match the SAN in the cert we are using
-						Hostname: "server.default.svc",
-					},
-					// Do not inject, as we are testing non-Istio TLS here
-					Subsets: []echo.SubsetConfig{{
-						Version:     "v1",
-						Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
-					}},
-				}).
-				BuildOrFail(t)
+						// TODO rebase on main multicluster testing branch, these are unused anyway
+						// Pilot: pilots[c.Index()],
+						// Set up TLS certs on the server. This will make the server listen with these credentials.
+						TLSSettings: serverTLSSettings,
+						// Do not inject, as we are testing non-Istio TLS here
+						Subsets: []echo.SubsetConfig{{
+							Version:     "v1",
+							Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
+						}},
+					})
+			}
+			builder.BuildOrFail(t)
 
-			for _, tt := range []string{"grpc", "http", "tcp"} {
-				t.Run(tt, func(t *testing.T) {
-					retry.UntilSuccessOrFail(ctx, func() error {
-						opts := echo.CallOptions{
-							Target:   server,
-							PortName: tt,
-						}
-						if tt == "tcp" {
-							opts.Scheme = scheme.TCP
-						}
-						resp, err := client.Call(opts)
-						if err != nil {
-							return err
+			for _, src := range env.Clusters() {
+				for _, dst := range env.Clusters() {
+					client, server := svcs[src.Index()][0], svcs[dst.Index()][1]
+					ctx.NewSubTest(fmt.Sprintf("%d->%d", src.Index(), dst.Index())).Run(func(ctx framework.TestContext) {
```

Comment:
What is the use/implication of using NewSubTest vs running tests on the main test context?

Reply:
In this instance, I guess none for the most part. For the most part it is good practice so if resources are created they can be automatically cleaned up/have data dumped on failure within that subcontext.

Reply:
Ah got it, I couldn't get Feature flag to work with NewSubTest so I'll watch out for that check. Thanks for the additional context.